### PR TITLE
Update gradle to build TFLilte Java bindings AAR library

### PR DESCRIPTION
**Pull request by KoolJBlack**
Process for building the AAR:

1. Start AndroidStudio. Select "Open File or Project" then choose `iree/bindings/tflite/java/gragle.build`.
2. AndroidStudio should sync the project and setup gradlew uner `iree/bindings/tflite/java`.
3. Make the project using android studio or run the build directly in terminal:
```shell
./gradlew build
```

This produces two libraries under `iree/bindings/tflite/java/build/outputs/aar`: 
* `iree-tflite-bindings-debug.aar` 
*  `iree-tflite-bindings-release.aar` 

AAR contents:
Screen Shot 2021-06-14 at 2 42 38 PM

Include either library in another gradle project for iree tflite binding support. 

Addresses #6198

### CODE REVIEW

**Review by not-jenni — APPROVED** [review]

**Comment by not-jenni** [comment]
Thank you for the instructions on how to build the AAR!

File: bindings/tflite/java/build.gradle [hunk 1]
```diff
@@ -48,9 +47,12 @@ android {
...
         main {
-            java.srcDirs = ['java/org/tensorflow/lite']
-            jni.srcDirs = ['java/org/tensorflow/lite/native']
-            manifest.srcFile 'org/tensorflow/lite/tests/TestManifest.xml'
+            manifest.srcFile 'org/tensorflow/lite/AndroidManifest.xml'
+            java {
+                srcDirs('org/tensorflow/lite')
+                exclude('tests/**') // Don't build tests into the library
+            }
+            jni.srcDirs = ['org/tensorflow/lite/native']
```
Comment (ScottTodd) [inline bindings/tflite/java/build.gradle:55]: Are these actually being included? Your screenshot of the .aar shows shared objects from https://github.com/google/iree/tree/main/iree/tools/android/run_module_app , which we _don't_ want in the archive. I'm actually pretty surprised this configuration is managing to pull sources from that directory.

**Review by ScottTodd — CHANGES_REQUESTED** [review]

**Comment by ScottTodd** [comment]
> Thank you for the instructions on how to build the AAR!

Agreed :) would be even better to include them in a README or file under https://github.com/google/iree/tree/main/docs/developers

File: bindings/tflite/java/org/tensorflow/lite/native/CMakeLists.txt [hunk 2]
```diff
@@ -6,17 +6,43 @@
 
 iree_add_all_subdirs()
 
-iree_cc_library(
-  NAME
-    native_deps
-  SRCS
+# iree_cc_library(
```
Comment (ScottTodd) [inline bindings/tflite/java/org/tensorflow/lite/native/CMakeLists.txt]: can remove this (or keep as reference, but then it could go out of sync)

**Review by ScottTodd — APPROVED** [review]
👍

**Pull request merged**